ocaml/typing/typemod.ml

```diff
@@ -3697,15 +3697,19 @@ let type_implementation ~sourcefile outputprefix modulename initial_env ast =
       if !Clflags.as_parameter then
         error Cannot_compile_implementation_as_parameter;
       register_params !Clflags.parameters;
-      let (str, sg, names, shape, finalenv) =
-        Profile.record_call "infer" (fun () ->
-          type_structure ~expected_sig:None initial_env ast) in
-      let uid = Uid.of_compilation_unit_id modulename in
-      let shape = Shape.set_uid_if_none shape uid in
-      if !Clflags.binary_annotations_cms then
-        cms_register_toplevel_struct_attributes ~sourcefile ~uid ast;
-      let simple_sg = Signature_names.simplify finalenv names sg in
+      let process_structure expected_sig =
```

Comment:
I think it might be better to call this `type_structure`, which shadows the existing `type_structure`. This prevents the rest of the function calls the existing `type_structure`.